pkg/knuu/knuu_test.go

```diff
@@ -66,70 +66,84 @@ func TestNew(t *testing.T) {
 	}{
 		{
 			name:        "Default initialization",
-			options:     Options{},
+			options:     Options{TestScope: "test"},
 			expectError: false,
 			validateFunc: func(t *testing.T, k *Knuu) {
 				assert.NotNil(t, k)
 				assert.NotNil(t, k.Logger)
 				assert.NotNil(t, k.K8sClient)
-				assert.NotNil(t, k.MinioClient)
 				assert.NotNil(t, k.ImageBuilder)
 				assert.Equal(t, defaultTimeout, k.timeout)
 			},
 		},
 		{
-			name: "With custom Logger",
+			name:        "With custom Minio client without setting k8sClient",
+			options:     Options{MinioClient: &minio.Minio{}},
+			expectError: true,
```

Comment:
To help catch some of the logic errors we are uncovering in this PR, I'd recommend you update this test struct to accept the actual error, not just whether you expect an error.